### DISCUSSION TITLE
Switch network on WalletConnect / Trust Wallet

### DISCUSSION THREAD

**Discussion by ilterugur** [wevm/wagmi#595] [Q&A]
Hello. How can i trigger a network switch on Trust Wallet? switchNetwork on useNetwork returns undefined. web3React v8 beta supports it somehow.

**Comment by tmm** [comment]
Are you using WalletConnect? Not all wallets support programmatic chain switching so they need to be added to the list [here](https://github.com/tmm/wagmi/blob/main/packages/core/src/connectors/walletConnect.ts#L16-L22=) in order for `switchNetwork` to work.

I just downloaded Trust Wallet and added it to the list, but wasn't able to switch chains programmatically. Also, I took a quick look at the [developer site](https://developer.trustwallet.com/) and couldn't find anything about chain switching.

**Comment by ilterugur** [comment]
> Are you using WalletConnect? Not all wallets support programmatic chain switching so they need to be added to the list [here](https://github.com/tmm/wagmi/blob/main/packages/core/src/connectors/walletConnect.ts#L16-L22=) in order for `switchNetwork` to work.
> 
> I just downloaded Trust Wallet and added it to the list, but wasn't able to switch chains programmatically. Also, I took a quick look at the [developer site](https://developer.trustwallet.com/) and couldn't find anything about chain switching.

Yeah i know but like i said web3-react can do it somehow and its quite interesting. Most of my customers force me to implement chain switching on my projects. I really love wagmi and i want to use it on all of my projects but it doesnt offer this feature unlike web3-react

**Reply by tmm** [reply]
Can you post the code you use to do this in web3-react?

**Reply by ilterugur** [reply]
> Can you post the code you use to do this in web3-react?

https://web3-react-mu.vercel.app/

https://github.com/NoahZinsmeister/web3-react/tree/main/packages%2Fexample-next

I thought it was using WalletConnect but now i see its connecting through metamask connector somehow.

**Reply by tmm** [reply]
Are you using Trust Wallet's built-in browser?

**Reply by ilterugur** [reply]
> Are you using Trust Wallet's built-in browser?

Yes

Btw how can i make it connect automatically to injected connector if its trust wallet? As i remember, it doesnt connect automatically on web3-react example above on desktop but it does on trustwallet browser.
